File: code_examples/3-Structured_Problem_Solving/1-Variables/CSharp/variables.cs
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program shows declaring variables
*/

using System;

/*
 * The main class
 * Contains all methods for performing basic variable usage
*/
class MainClass {
  public static void Main (string[] args) {
    // variable definition
    bool isCurrent = true;               // bool
    int age = 32;                        // int
    float area = 32.45F;                 // float
    string someWords = "Hello, World!";  // string

    Console.WriteLine (isCurrent);
    Console.WriteLine (age);
    Console.WriteLine (area);
    Console.WriteLine (someWords);
  }
}

File: code_examples/3-Structured_Problem_Solving/1-Variables/Java/Variables.java
/*
 * This program shows declaring variables
 *
 * @author  Mr Coxall
 * @version 1.0
 * @since   2020-09-01
*/

final class Variables {
    private Variables() {
        // Prevent instantiation
        // Optional: throw an exception e.g. AssertionError
        // if this ever *is* called
        throw new IllegalStateException("Cannot be instantiated");
    }
    /**
     * Main entry point into program.
     * @param args nothing passed in
    */
    public static void main(final String[] args) {
        // variable definition
        boolean isCurrent = true;
        int age = 0;
        float area = 14.65F;
        String someWords = "Hello, World!";

        System.out.println(isCurrent);
        System.out.println(age);
        System.out.println(area);
        System.out.println(someWords);

        System.out.println("\nDone.");
    }
}
